2.11 release notes (#948)
kyma-project/website: content/blog-posts/2023-02-09-release-notes-2.11/index.md

---
title: "Kyma 2.11"
author:
  name: "Zhoujing Wang, RM @Kyma, and Natalia Sitko, Technical Writer @Kyma"
tags:
  - release-notes 
type: release 
releaseTag: "2.11"
redirectFrom:
  - "/blog/release-notes-211"
---

> With love for cloud-native technologies in our hearts and lots of cool ideas in our minds, we've got through the middle of this snowy and cloudy winter season. Now that we have reached the 2.11 milestone, we can share what we accomplished during this part of the journey. Version 2.11 of Kyma comes with a few exciting improvements, such as FluentBit DaemonSet fully managed by the operator, Istio and FLuentBit upgrades, the new finalizer for APIRule CR, and that is by no means all. Read on to find out the rest of the news.

Reviewer: ```suggestion
With love for cloud-native technologies in our hearts and lots of cool ideas in our minds, we've got through the middle of this snowy and cloudy winter season. Now that we have reached the 2.11 milestone, we can share what we accomplished during this part of the journey. Version 2.11 of Kyma comes with a few exciting improvements, such as FluentBit DaemonSet fully managed by the operator, Istio and FluentBit upgrades, the new finalizer for APIRule CR, and that is by no means all. Read on to find out the rest of the news.
```

Reviewer 2: I think `Fluent Bit` officially uses a blank space in their name instead of camel case, see https://fluentbit.io/

Author: You're right, thank you. I'll correct the RNs.

> With love for cloud-native technologies in our hearts and lots of cool ideas in our minds, we've got through the middle of this snowy and cloudy winter season. Now that we have reached the 2.11 milestone, we can share what we accomplished during this part of the journey. Version 2.11 of Kyma comes with a few exciting improvements, such as FluentBit Deamon Set fully managed by the operator, Istio and FLuentBit upgrades, the new finalizer for APIRule CR, and that is by no means all. Read on to find out the rest of the news.

Reviewer 3: ```suggestion
With love for cloud-native technologies in our hearts and lots of cool ideas in our minds, we've got through the middle of this snowy and cloudy winter season. Now that we have reached the 2.11 milestone, we can share what we accomplished during this part of the journey. Version 2.11 of Kyma comes with a few exciting improvements, such as FluentBit DaemonSet fully managed by the operator, Istio and FLuentBit upgrades, the new finalizer for APIRule CR, and that is by no means all. Read on to find out the rest of the news.
```
DaemonSet is a Kubernetes concept, so it should be written in camelCase. See [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) and [our guides](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#camelcase) for reference.

<!-- overview -->

See the overview of all changes in this release:

- [Application Connectivity](#application-connectivity) - **encodeUrl** parameter
- [Observability](#observability) – Kiali removed, Jaeger deprecated

> - [Telemetry](#telemetry) – FluentBit upgraded to version 2.0.8, FluentBit DaemonSet fully managed by the operator, system traces filtering, TracePipelines extension

Reviewer: ```suggestion
- [Telemetry](#telemetry) – FluentBit upgraded to version 2.0.8, FluentBit DaemonSet fully managed by the operator, improvements for TracePipeline
```

> - [Telemetry](#telemetry) – FluentBit upgraded to version 2.0.8, FluentBit Deamon Set fully managed by the operator, system traces filtering, TracePipelines extension

Reviewer 3: ```suggestion
- [Telemetry](#telemetry) – FluentBit upgraded to version 2.0.8, FluentBit DaemonSet fully managed by the operator, system traces filtering, TracePipelines extension
```
See my previous comment.

- [API Gateway](#api-gateway) – improved status handling and CPU reconciling, new finalizer for APIRule CR, APIRule CR conversion bug fixed
- [Security](#security) – Istio upgraded to version 1.16.2, PodPreset functionality removed
- [Eventing](#eventing) – NATS image updated to version 2.9.11

## Application Connectivity 
With Kyma 2.11, we introduced the **encodeUrl** parameter, which allows you to control whether Application Gateway encodes URL characters. Set the parameter to `false` so that the URL path remains unchanged or to `true` to enable the encoding. For example, if the **encodeUrl** is set to `true`, the `/app/api/sample(1%2C2%2C3)%2FescapingURL` path changes to `/app/api/sample%281%2C2%2C3%29/escaping`. When it is set to `false`, the `/app/api/sample(1%2C2%2C3)%2Fescaping` path stays intact. 

## Observability

### Kiali 
Kiali was deprecated with Kyma 2.8 and removed with Kyma 2.11. If you want to continue using it, follow the steps in the [Kiali example](https://github.com/kyma-project/examples/tree/main/kiali) to deploy your custom Kiali. 

### Tracing 

> The tracing component was deprecated and will be removed with Kyma 2.14. If you want to continue using Jaeger, follow the steps in the [Jaeger example](https://github.com/kyma-project/examples/tree/main/jaeger) to deploy your custom Jaeger. 

Reviewer: ```suggestion
The tracing component is deprecated and will be removed with Kyma 2.14. If you want to continue using Jaeger, follow the steps in the [Jaeger example](https://github.com/kyma-project/examples/tree/main/jaeger) to deploy your custom Jaeger. 
```
It gets deprecated by that sentence here, it is the actual deprecation note!

Author: Thanks for the comment. I changed the sentence.

 
## Telemetry

### Logging 

> To run the Telemetry module as a fully self-contained solution with all the resources managed by the module operator, we introduced a series of changes and improvements. With Kyma 2.11, we completed the process and released FluentBit DaemonSet [fully managed by the operator](https://github.com/kyma-project/kyma/issues/16570). From now, the FluentBit installation is responsible for serving LogPipeline resources and can be provisioned on-demand only if the user requests a LogPipeline. The FluentBit settings are managed by the module operator and can no longer be influenced directly using the module Helm values. 

Reviewer: ```suggestion
To run the Telemetry module as a fully self-contained solution with all the resources managed by the module operator, we introduced a series of changes and improvements. With Kyma 2.11, we completed the process and released FluentBit DaemonSet [fully managed by the operator](https://github.com/kyma-project/kyma/issues/16570). From now, the FluentBit installation can be provisioned on-demand only if the user requests a LogPipeline. The FluentBit settings are managed by the module operator and can no longer be influenced directly using the module Helm values. 
```

> To be able to run the telemetry module as a fully self-contained solution with all the resources managed by the module operator, we introduced a series of changes and improvements. With Kyma 2.11, we completed the process and released FluentBit Daemon Set [fully managed by the operator](https://github.com/kyma-project/kyma/issues/16570). From now, the FluentBit installation is responsible for serving LogPipeline resources and can be provisioned on-demand only if the user requests a LogPipeline. The FluentBit settings are managed by the module operator and can no longer be influenced directly using the module Helm values. 

Reviewer 3: ```suggestion
To run the Telemetry module as a fully self-contained solution with all the resources managed by the module operator, we introduced a series of changes and improvements. With Kyma 2.11, we completed the process and released FluentBit DaemonSet [fully managed by the operator](https://github.com/kyma-project/kyma/issues/16570). From now, the FluentBit installation is responsible for serving LogPipeline resources and can be provisioned on-demand only if the user requests a LogPipeline. The FluentBit settings are managed by the module operator and can no longer be influenced directly using the module Helm values. 
```
I think that when we talk about a module, we should use uppercase. See [Telemetry in Kyma](https://kyma-project.io/docs/kyma/latest/01-overview/main-areas/telemetry/) for reference.

> The FluentBit Daemon Set [got upgraded to the new major version 2](https://github.com/kyma-project/kyma/issues/15932). Besides the performance improvements, the 2.0.8 version introduces OTLP support and enables the Kyma team to start working on the official [OTLP support for LogPipelines](https://github.com/kyma-project/kyma/issues/16307).

Reviewer 3: ```suggestion
The FluentBit DaemonSet [got upgraded to the new major version 2](https://github.com/kyma-project/kyma/issues/15932). Besides the performance improvements, the 2.0.8 version introduces OTLP support and enables the Kyma team to start working on the official [OTLP support for LogPipelines](https://github.com/kyma-project/kyma/issues/16307).
```

We fixed a bug so that the [log files are always read from the beginning](https://github.com/kyma-project/kyma/issues/16645) when starting a FluentBit instance on a newly provisioned node.

### Traces 

> System traces emitted by Istio, for example, while communicating with the trace-collector, are now [filtered out by default](https://github.com/kyma-project/kyma/issues/16514). We also added the [extension for TracePipelines](https://github.com/kyma-project/kyma/issues/16393) to Kyma Dashboard and fixed a [bug](https://github.com/kyma-project/kyma/issues/16531) that caused the TracePipeline status not to reflect the status of the managed trace-collector properly. With version 2.11 of Kyma, we introduced support for [custom headers on the OTLP output](https://github.com/kyma-project/kyma/issues/16554) that enables integrations to providers with native OTLP support, such as [Dynatrace](https://www.dynatrace.com/support/help/extend-dynatrace/opentelemetry#tabgroup--opentelemetry--instrument-without-oneagent). 

Reviewer: ```suggestion
System traces emitted by Istio, for example, while communicating with the trace-collector, are now [filtered out by default](https://github.com/kyma-project/kyma/issues/16514). We also added the [extension for TracePipelines](https://github.com/kyma-project/kyma/issues/16393) to Kyma Dashboard and fixed a [bug](https://github.com/kyma-project/kyma/issues/16531) that caused the TracePipeline status not to reflect the status of the managed trace-collector properly.

With version 2.11 of Kyma, we introduced support for [custom headers on the OTLP output](https://github.com/kyma-project/kyma/issues/16554) that enables integrations to providers with native OTLP support, such as [Dynatrace](https://www.dynatrace.com/support/help/extend-dynatrace/opentelemetry#tabgroup--opentelemetry--instrument-without-oneagent). 
```

> System traces emitted by Istio, for example, while communicating with the trace-collector, are now [filtered out by default](https://github.com/kyma-project/kyma/issues/16514). We also added the [extension for TracePipelines](https://github.com/kyma-project/kyma/issues/16393)to Kyma Dashboard and fixed a [bug](https://github.com/kyma-project/kyma/issues/16531) that caused the TracePipeline status not to reflect the status of the managed trace-collector properly. With version 2.11 of Kyma, we introduced support for [custom headers on the OTLP output](https://github.com/kyma-project/kyma/issues/16554) that enables integrations to providers with native OTLP support, such as [Dynatrace](https://www.dynatrace.com/support/help/extend-dynatrace/opentelemetry#tabgroup--opentelemetry--instrument-without-oneagent). 

Reviewer 3: ```suggestion
System traces emitted by Istio, for example, while communicating with the trace-collector, are now [filtered out by default](https://github.com/kyma-project/kyma/issues/16514). We also added the [extension for TracePipelines](https://github.com/kyma-project/kyma/issues/16393) to Kyma Dashboard and fixed a [bug](https://github.com/kyma-project/kyma/issues/16531) that caused the TracePipeline status not to reflect the status of the managed trace-collector properly. With version 2.11 of Kyma, we introduced support for [custom headers on the OTLP output](https://github.com/kyma-project/kyma/issues/16554) that enables integrations to providers with native OTLP support, such as [Dynatrace](https://www.dynatrace.com/support/help/extend-dynatrace/opentelemetry#tabgroup--opentelemetry--instrument-without-oneagent). 
```

 
 
## API Gateway

With this version of API Gateway, we improved the status handling, added the finalizer for APIRule CR, fixed the bug related to APIRule CR conversion, and resolved the reconciling CPU issue for Azure.

## Security
 
### Service Mesh 
With Kyma 2.11, we upgraded Istio 1.16.1 to version 1.16.2. Read the official [Istio 1.16.2 release notes](https://istio.io/latest/news/releases/1.16.x/announcing-1.16/upgrade-notes/) to learn about the changes introduced with the new Istio release.

> ### Pod Preset 

Reviewer 4: ```suggestion
### Pod Presets
```
